### ISSUE TITLE
Low Correlation Scores

### ISSUE THREAD

**Issue by jianzhitaiyouqule** [proto17/dji_droneid#44]
I used a sample file from [https://github.com/RUB-SysSec/DroneSecurity](url),and then i changed the paramters of process_file.m like this:
![fb1165944a7a4d421e928b658ea0a43](https://github.com/proto17/dji_droneid/assets/113488528/2a601335-9d9a-424b-8fa5-0ebfa2b968f4)
i got low Correlation Scores.

Could someone help me?

**Comment by jianzhitaiyouqule** [comment]
Low Correlation Scores
![9b79d4481633663c2f6c07eefffc838](https://github.com/proto17/dji_droneid/assets/113488528/8827a4d1-60aa-4b59-8dba-6b8d872b7895)

**Comment by jianzhitaiyouqule** [comment]
![95b3ec76062af4ca31d2d62e6faa319](https://github.com/proto17/dji_droneid/assets/113488528/8e13be36-7ea8-43d9-b288-250343bce20a)
Is this DroneId?

**Comment by proto17** [comment]
The `mavic_air_2` file appears to be sampled at 50 MSPS which is not a valid rate for `process_file.m` (must be a multiple of 15.36 MSPS).  Additionally, just looking at the signal in Baudline, it appears that there is a 2 MHz offset assuming a 50 MSPS sample rate.  So, you'll need to resample the file from 50 MSPS to 15.36 MSPS or 30.72 MSPS, then change the `file_freq_offset` parameter to `2e6` and set `file_sample_rate` to the resampled rate.  I resampled the file with the script below, and then fed that resampled file into `process_file.m` using a sample rate of `30.72e6` and frequency offset of `-2e6` (to undo the 2 MHz offset in the file).  From there I saw good constellation points, but a failed CRC.  The demodulated bytes look good, but the CRC does not pass.  I haven't put it through any kind of parser to see if the data is actually valid.

There are multiple bursts present, but you would need to resample to 61.44 MSPS before feeding into `process_file.m` to be able to demodulate those using the `file_freq_offset` parameter.

```
file = '/tmp/mavic_air_2';
samples = read_complex(file, 0, Inf, 'single');

original_sample_rate = 50e6;
target_sample_rate = 30.72e6;

resampled = resample(samples, target_sample_rate, original_sample_rate);

figure(1);
subplot(3, 1, 1); plot(10 * log10(abs(fftshift(fft(samples)).^2))); title('Original File');
subplot(3, 1, 2); plot(10 * log10(abs(fftshift(fft(resampled)).^2))); title('Resampled File');

% Just appending `.resampled` to the file name
output_name = [file, '.resampled'];
handle = fopen(output_name, "w");
fwrite(handle, reshape([real(resampled), imag(resampled)].', [], 1), 'single');
fclose(handle);

subplot(3, 1, 3); plot(10 * log10(abs(fftshift(fft(read_complex(output_name, 0, Inf, 'single'))).^2)));
title('Read Back Resampled File');
```

**Comment by jianzhitaiyouqule** [comment]
I resampled the file from 50 MSPS to 30.72 MSPS, just as you said，there is a CRC error.
![1693981669031](https://github.com/proto17/dji_droneid/assets/113488528/6ee1ba9c-a9c0-47c1-b3c0-a3f06f99c129)
Then i resampled the file to 61.44 MSPS, I got two different frames.
![1693982132263](https://github.com/proto17/dji_droneid/assets/113488528/ce71e336-b010-4d60-b2d8-38da3fe4d48f)

I'm trying to decode it to get the information by using dji_decode.py from dragon OS. I found it from the comments.

Thank you very much for your reply, you made me realize that I overlooked a lot of details,really appreciate your assistance.😀
